firebase-firestore/src/main/java/com/google/firebase/firestore/model/TargetIndexMatcher.java

```diff
@@ -192,6 +192,59 @@ public boolean servedByIndex(FieldIndex index) {
     return true;
   }
 
+  /** Returns a full matched field index for this target. */
+  public FieldIndex BuildTargetIndex() {
```

Comment:
nit: The first character of `BuildTargetIndex` should be lowercase, conventionally.